### ISSUE TITLE
Not working with R kernel without lib paths

### ISSUE THREAD

**Issue by roaldarbol** [renan-r-santos/pixi-kernel#15]
Hi Renan! Really awesome that you're getting the multi-kernel support up and running! 

I'm having issues getting the R kernel working without specifying the R_LIBS/R_LIBS_USER paths first. I think you should probably be able to use the kernel even without manually having to specify that yourself. Do you think there could be a way around that?
Otherwise, I think it would be worth documenting.
 
Steps to reproduce:
```bash
pixi init
pixi add jupyter pixi-kernel r-irkernel r-base
pixi run jupyter lab
```
Then start a notebook with a Pixi-R kernel. I get the following output:

```python
[I 2024-04-12 20:16:28.141 ServerApp] [pixi-kernel] found Pixi 0.19.1
[I 2024-04-12 20:16:28.177 ServerApp] [pixi-kernel] project manifest path: /Users/roaldarbol/Downloads/test-jupy/pixi.toml
[I 2024-04-12 20:16:28.178 ServerApp] [pixi-kernel] launching kernel: ['pixi', 'run', '--manifest-path=/Users/roaldarbol/Downloads/test-jupy/pixi.toml', 'R', '--slave', '-e', "'IRkernel::main()'", '--args', '/Users/roaldarbol/Library/Jupyter/runtime/kernel-8327278f-a84f-4ed3-81b4-13999f630639.json']
[I 2024-04-12 20:16:28.208 ServerApp] Kernel started: 8327278f-a84f-4ed3-81b4-13999f630639
Warning: namespace ‘pbdZMQ’ is not available and has been replaced
by .GlobalEnv when processing object ‘.pbd_env’
Error in library.dynam(lib, package, package.lib) : 
  shared object ‘digest.dylib’ not found
Calls: loadNamespace -> namespaceImport -> loadNamespace -> library.dynam
Execution halted
[I 2024-04-12 20:16:34.188 ServerApp] AsyncIOLoopKernelRestarter: restarting kernel (1/5), new random ports
[I 2024-04-12 20:16:34.228 ServerApp] [pixi-kernel] found Pixi 0.19.1
[I 2024-04-12 20:16:34.259 ServerApp] [pixi-kernel] project manifest path: /Users/roaldarbol/Downloads/test-jupy/pixi.toml
[I 2024-04-12 20:16:34.259 ServerApp] [pixi-kernel] launching kernel: ['pixi', 'run', '--manifest-path=/Users/roaldarbol/Downloads/test-jupy/pixi.toml', 'R', '--slave', '-e', "'IRkernel::main()'", '--args', '/Users/roaldarbol/Library/Jupyter/runtime/kernel-8327278f-a84f-4ed3-81b4-13999f630639.json']
Warning: namespace ‘pbdZMQ’ is not available and has been replaced
by .GlobalEnv when processing object ‘.pbd_env’
Error in library.dynam(lib, package, package.lib) : 
  shared object ‘digest.dylib’ not found
Calls: loadNamespace -> namespaceImport -> loadNamespace -> library.dynam
Execution halted
[I 2024-04-12 20:16:40.271 ServerApp] AsyncIOLoopKernelRestarter: restarting kernel (2/5), new random ports
[I 2024-04-12 20:16:40.300 ServerApp] [pixi-kernel] found Pixi 0.19.1
[I 2024-04-12 20:16:40.333 ServerApp] [pixi-kernel] project manifest path: /Users/roaldarbol/Downloads/test-jupy/pixi.toml
[I 2024-04-12 20:16:40.333 ServerApp] [pixi-kernel] launching kernel: ['pixi', 'run', '--manifest-path=/Users/roaldarbol/Downloads/test-jupy/pixi.toml', 'R', '--slave', '-e', "'IRkernel::main()'", '--args', '/Users/roaldarbol/Library/Jupyter/runtime/kernel-8327278f-a84f-4ed3-81b4-13999f630639.json']
Warning: namespace ‘pbdZMQ’ is not available and has been replaced
by .GlobalEnv when processing object ‘.pbd_env’
Error in library.dynam(lib, package, package.lib) : 
  shared object ‘digest.dylib’ not found
Calls: loadNamespace -> namespaceImport -> loadNamespace -> library.dynam
Execution halted
[I 2024-04-12 20:16:46.343 ServerApp] AsyncIOLoopKernelRestarter: restarting kernel (3/5), new random ports
[I 2024-04-12 20:16:46.388 ServerApp] [pixi-kernel] found Pixi 0.19.1
[I 2024-04-12 20:16:46.422 ServerApp] [pixi-kernel] project manifest path: /Users/roaldarbol/Downloads/test-jupy/pixi.toml
[I 2024-04-12 20:16:46.422 ServerApp] [pixi-kernel] launching kernel: ['pixi', 'run', '--manifest-path=/Users/roaldarbol/Downloads/test-jupy/pixi.toml', 'R', '--slave', '-e', "'IRkernel::main()'", '--args', '/Users/roaldarbol/Library/Jupyter/runtime/kernel-8327278f-a84f-4ed3-81b4-13999f630639.json']
Warning: namespace ‘pbdZMQ’ is not available and has been replaced
by .GlobalEnv when processing object ‘.pbd_env’
Error in library.dynam(lib, package, package.lib) : 
  shared object ‘digest.dylib’ not found
Calls: loadNamespace -> namespaceImport -> loadNamespace -> library.dynam
Execution halted
[I 2024-04-12 20:16:52.434 ServerApp] AsyncIOLoopKernelRestarter: restarting kernel (4/5), new random ports
[I 2024-04-12 20:16:52.476 ServerApp] [pixi-kernel] found Pixi 0.19.1
[I 2024-04-12 20:16:52.508 ServerApp] [pixi-kernel] project manifest path: /Users/roaldarbol/Downloads/test-jupy/pixi.toml
[I 2024-04-12 20:16:52.508 ServerApp] [pixi-kernel] launching kernel: ['pixi', 'run', '--manifest-path=/Users/roaldarbol/Downloads/test-jupy/pixi.toml', 'R', '--slave', '-e', "'IRkernel::main()'", '--args', '/Users/roaldarbol/Library/Jupyter/runtime/kernel-8327278f-a84f-4ed3-81b4-13999f630639.json']
Warning: namespace ‘pbdZMQ’ is not available and has been replaced
by .GlobalEnv when processing object ‘.pbd_env’
Error in library.dynam(lib, package, package.lib) : 
  shared object ‘digest.dylib’ not found
Calls: loadNamespace -> namespaceImport -> loadNamespace -> library.dynam
Execution halted
[I 2024-04-12 20:16:58.520 ServerApp] AsyncIOLoopKernelRestarter: restarting kernel (5/5), new random ports
[I 2024-04-12 20:16:58.561 ServerApp] [pixi-kernel] found Pixi 0.19.1
[I 2024-04-12 20:16:58.592 ServerApp] [pixi-kernel] project manifest path: /Users/roaldarbol/Downloads/test-jupy/pixi.toml
[I 2024-04-12 20:16:58.592 ServerApp] [pixi-kernel] launching kernel: ['pixi', 'run', '--manifest-path=/Users/roaldarbol/Downloads/test-jupy/pixi.toml', 'R', '--slave', '-e', "'IRkernel::main()'", '--args', '/Users/roaldarbol/Library/Jupyter/runtime/kernel-8327278f-a84f-4ed3-81b4-13999f630639.json']
Warning: namespace ‘pbdZMQ’ is not available and has been replaced
by .GlobalEnv when processing object ‘.pbd_env’
Error in library.dynam(lib, package, package.lib) : 
  shared object ‘digest.dylib’ not found
Calls: loadNamespace -> namespaceImport -> loadNamespace -> library.dynam
Execution halted
[W 2024-04-12 20:17:04.603 ServerApp] AsyncIOLoopKernelRestarter: restart failed
[W 2024-04-12 20:17:04.603 ServerApp] Kernel 8327278f-a84f-4ed3-81b4-13999f630639 died, removing from map.
[W 2024-04-12 20:17:28.229 ServerApp] Timeout waiting for kernel_info reply from 8327278f-a84f-4ed3-81b4-13999f630639
```

**Comment by renan-r-santos** [comment]
Hi @roaldarbol, thanks for the issue report.
The [mybinder demo](https://mybinder.org/v2/gh/renan-r-santos/pixi-kernel-binder/main?labpath=example.ipynb) works fine for me using the `Pixi - R` kernel, so even though I don't know much about R, I believe either you have a global installation of R that's interfering with the kernel or adding `r-irkernel` and `r-base` to the Jupyter environment is also interfering.

Note that you don't need to install `r-irkernel` and `r-base` with `pixi-kernel` but you do need to create a new `pixi` project inside JupyterLab and add `i-irkernel` there, see the [quick start docs](https://github.com/renan-r-santos/pixi-kernel?tab=readme-ov-file#quick-start).

Nonetheless, those other R installations weren't supposed to break the `Pixi - R` kernel, so I'm proposing a fix at https://github.com/renan-r-santos/pixi-kernel/pull/16. Would you be able to try it, by cloning, checking out the branch and doing an editable install? I've had a few problems with editable installs though, so you might need to delete the `rattler` cache at `~/.cache/rattler/cache` and `.pixi` and `pixi.lock` folders and file.

The fix sets `R_LIBS`, `R_LIBS_SITE` and `R_LIBS_USER` to the path of the Pixi manifest file that is running the `Pixi - R` kernel, so I believe this should be a robust solution regardless of other R installations, see https://github.com/IRkernel/IRkernel/issues/583#issuecomment-480721221.

**Comment by roaldarbol** [comment]
> Note that you don't need to install `r-irkernel` and `r-base` with `pixi-kernel` but you do need to create a new `pixi` project inside JupyterLab and add `i-irkernel` there, see the [quick start docs](https://github.com/renan-r-santos/pixi-kernel?tab=readme-ov-file#quick-start).

Ah, I may be confused here. Do I need to add `r-irkernel` within Jupyter? Can't it be done from in the pixi project before opening Jupyter Lab?

I should not be having an interfering R installation, but it generally points to the global library rather than the project-based one - so it will not be picking up the pixi-installed libraries by default. So I think your patch should work - I've had the exact same issues trying to install R packages from conda-forge within a pixi project - as an aside, I've created an R package, https://github.com/roaldarbol/r-pixi, that allows installing R packages within R using pixi (which is the normal R workflow) to circumvent those issues for regular users.
I won't be able to test out your patch today, but can maybe have a stab at it tomorrow.

**Comment by renan-r-santos** [comment]
I published a new release `0.2.1` with the changes to R environment variables. It is on PyPI and soon on conda-forge. It would be awesome if you could try it. Thanks.

**Comment by roaldarbol** [comment]
Works beautifully! Well done!
